internal/command/workspace_command_test.go

```diff
@@ -16,11 +16,159 @@ import (
 	"github.com/hashicorp/terraform/internal/backend"
 	"github.com/hashicorp/terraform/internal/backend/local"
 	"github.com/hashicorp/terraform/internal/backend/remote-state/inmem"
+	"github.com/hashicorp/terraform/internal/providers"
 	"github.com/hashicorp/terraform/internal/states"
 	"github.com/hashicorp/terraform/internal/states/statefile"
 	"github.com/hashicorp/terraform/internal/states/statemgr"
 )
 
+func TestWorkspace_allCommands_pluggableStateStore(t *testing.T) {
+	// Create a temporary working directory with pluggable state storage in the config
+	td := t.TempDir()
+	testCopyDir(t, testFixturePath("state-store-new"), td)
+	t.Chdir(td)
+
+	mock := testStateStoreMockWithChunkNegotiation(t, 1000)
+	newMeta := func(provider providers.Interface) (meta Meta, ui *cli.MockUi, close func()) {
+		// Assumes the mocked provider is hashicorp/test
+		providerSource, close := newMockProviderSource(t, map[string][]string{
+			"hashicorp/test": {"1.2.3"},
+		})
+
+		ui = new(cli.MockUi)
+		view, _ := testView(t)
+		meta = Meta{
+			AllowExperimentalFeatures: true,
+			Ui:                        ui,
+			View:                      view,
+			testingOverrides: &testingOverrides{
+				Providers: map[addrs.Provider]providers.Factory{
+					addrs.NewDefaultProvider("test"): providers.FactoryFixed(mock),
+				},
+			},
+			ProviderSource: providerSource,
+		}
+		return meta, ui, close
+	}
+
+	//// Init
+	meta, ui, close := newMeta(mock)
+	defer close()
```

Comment:
Probably minor but what is the reason we keep re-defining the meta for every command?

I'm assuming `close()` in all the `defer`s still refer to each individual function but still looks a little bit confusing for a reader. Could we not just reuse the same one instance?